### PR TITLE
support podPool

### PR DESCRIPTION
**Pull request by lliranbabi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,31 +66,31 @@ To use the Kwok Operator to provision nodes, follow these steps:
     apiVersion: kwok.sigs.k8s.io/v1beta1
     kind: NodePool
     metadata:
-    labels:
-        app.kubernetes.io/instance: nodepool-sample
-    name: nodepool-sample
+      labels:
```
Comment (haimlevy2006) [inline README.md:69]: I think you have a lot extra leading spaces in this file

File: README.md [hunk 2]
```diff
@@ -130,14 +130,14 @@ To use the Kwok Operator to manage deployments and run the pods on top the nodes
...
    metadata:
-   labels:
-      app.kubernetes.io/name: deploymentpool
-      app.kubernetes.io/instance: deploymentpool-sample
-      app.kubernetes.io/part-of: kwok-operator
-      app.kubernetes.io/managed-by: kustomize
-      app.kubernetes.io/created-by: kwok-operator
-   name: deploymentpool-sample
-   namespace: default # -----> change if needed 
+      labels:
```
Comment (haimlevy2006) [inline README.md:133]: Same in this file

File: internal/controller/nodepool_controller_test.go [hunk 3]
```diff
@@ -147,12 +148,17 @@ func TestReconcileNodePool(t *testing.T) {
 
 	// delete the NodePool object and check if the nodes are deleted
 	err = fakeClient.Get(ctx, types.NamespacedName{Name: "single-nodepool"}, nodePool)
+	if err != nil {
+		log.Println("failed to get NodePool object")
+	}
 	err = fakeClient.Delete(ctx, nodePool)
+	if err == nil {
```
Comment (haimlevy2006) [inline internal/controller/nodepool_controller_test.go:155]: Should be "if err != nil"

Comment (lliranbabi) [reply internal/controller/nodepool_controller_test.go:155]: Nope , it should be like that , if the error equal nil it means the nodePool still exist and it's not what i'm expecting

File: README.md (hunk 2 above)
Comment (lliranbabi) [reply README.md:133]: it's fine

File: api/v1beta1/podpool_types.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Comment (yossig-runai) [inline api/v1beta1/podpool_types.go:24]: please remove these lines

File: internal/controller/nodepool_controller_test.go (hunk 3 above)
Comment (yossig-runai) [reply internal/controller/nodepool_controller_test.go:155]: what is the output if it succeed to delete?

File: internal/controller/podpool_controller.go [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+	if !controllerutil.ContainsFinalizer(podPool, controllerFinalizer) {
+		log.Info("Adding Finalizer for the PodPool")
+		err := r.addFinalizer(ctx, podPool)
+		if err != nil {
+			log.Error(err, "Failed to add finalizer to PodPool")
+			return ctrl.Result{}, err
+		}
+	}
+	// Get pods in the cluster with owner reference to the podPool
+
```
Comment (yossig-runai) [inline internal/controller/podpool_controller.go:91]: remove empty line

File: internal/controller/podpool_controller_test.go [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+			controllerReconciler := &PodPoolReconciler{
+				Client: k8sClient,
+				Scheme: k8sClient.Scheme(),
+			}
+
+			_, err := controllerReconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: typeNamespacedName,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			// TODO(user): Add more specific assertions depending on your controller's reconciliation logic.
```
Comment (yossig-runai) [inline internal/controller/podpool_controller_test.go:99]: remove the comment

File: config/samples/kwok.sigs_v1beta1_podpool.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+apiVersion: kwok.sigs.run-ai.com/v1beta1
+kind: PodPool
+metadata:
+  labels:
+    app.kubernetes.io/name: podpool
+    app.kubernetes.io/instance: podpool-sample
+    app.kubernetes.io/part-of: kwok-operator
+    app.kubernetes.io/created-by: kwok-operator
+  name: podpool-sample
```
Comment (dormullor) [inline config/samples/kwok.sigs_v1beta1_podpool.yaml:9]: Consider adding a namespace to this example

File: internal/controller/podpool_controller.go [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+	})
+
+	podAnnotation := podPool.Spec.PodTemplate.Annotations
+	if podAnnotation == nil {
+		podAnnotation = make(map[string]string)
+	}
+	podAnnotation[controllerAnnotation] = fakeString
+	for i := int32(len(pods)); i < podPool.Spec.PodCount; i++ {
+		// Create a new pod
+		pod := &corev1.Pod{
```
Comment (dormullor) [inline internal/controller/podpool_controller.go:299]: Why creating the pod object inside the loop if you are not using `i` ?

Comment (lliranbabi) [reply internal/controller/podpool_controller.go:298]: each one of them getting other name ... i'm generating the name of the pod with the name of the CRD

File: internal/controller/podpool_controller_test.go (hunk 6 above)
Comment (lliranbabi) [reply internal/controller/podpool_controller_test.go]: Done

**Review by haimlevy2006 — APPROVED** [review]

**Pull request merged**